File: app/src/main/java/com/commit451/gitlab/util/BypassImageGetterFactory.java
package com.commit451.gitlab.util;

import android.widget.TextView;

import com.commit451.bypasspicassoimagegetter.BypassPicassoImageGetter;
import com.commit451.gitlab.model.api.Project;
import com.squareup.picasso.Picasso;

import timber.log.Timber;

/**
 * Creates {@link BypassPicassoImageGetter}s which are configured to handle relative Urls
 */
public class BypassImageGetterFactory {

    public static BypassPicassoImageGetter create(TextView textView, Picasso picasso, final String baseUrl, final Project project) {
        BypassPicassoImageGetter getter = new BypassPicassoImageGetter(textView, picasso);
        getter.setSourceModifier(new BypassPicassoImageGetter.SourceModifier() {
            @Override
            public String modify(String source) {
                if (source.startsWith("/")) {
                    String url = baseUrl + "/" + project.getPathWithNamespace() + source;
                    Timber.d(url);
                    return url;
                }
                return source;
            }
        });
        return getter;
    }
}

File: app/src/main/java/com/commit451/gitlab/util/BypassImageGetterFactory.kt
package com.commit451.gitlab.util

import android.widget.TextView

import com.commit451.bypasspicassoimagegetter.BypassPicassoImageGetter
import com.commit451.gitlab.model.api.Project
import com.squareup.picasso.Picasso

import timber.log.Timber

/**
 * Creates [BypassPicassoImageGetter]s which are configured to handle relative Urls
 */
object BypassImageGetterFactory {

    fun create(textView: TextView, picasso: Picasso, baseUrl: String, project: Project): BypassPicassoImageGetter {
        val getter = BypassPicassoImageGetter(textView, picasso)
        getter.setSourceModifier(BypassPicassoImageGetter.SourceModifier { source ->
            if (source.startsWith("/")) {
                val url = baseUrl + "/" + project.pathWithNamespace + source
                Timber.d(url)
                return@SourceModifier url
            }
            source
        })
        return getter
    }
}
